src/Opaleye/Internal/Sql.hs

```diff
@@ -3,7 +3,7 @@
 
 module Opaleye.Internal.Sql where
 
-import           Prelude hiding (product)
+import           Prelude hiding (filter, product)
```

Comment:
Is this change needed?

Reply:
Oh, I see, it fixes a warning. That's fine.

Reply:
It's not "needed" here to fix the problems with `WITH`, but it does fix a name shadowing warning that I noticed while working on this. It's a separate commit from the other changes.

Reply:
Great, thanks for splitting it out as a separate commit.